Dockerfile

```diff
@@ -1,72 +1,68 @@
-    " \
-    && seq 1 8 | xargs -I{} mkdir -p /usr/share/man/man{} \
-    && apt-get update && apt-get install -y --no-install-recommends $RUN_DEPS \
-    && rm -rf /var/lib/apt/lists/*
+# Install dependencies in a virtualenv
+ENV VIRTUAL_ENV=/venv
 
-ADD requirements/ /requirements/
-RUN set -ex \
-    && BUILD_DEPS=" \
-        build-essential \
-        git \
-        libexpat1-dev \
-        libjpeg62-turbo-dev \
-        libpcre3-dev \
-        libpq-dev \
-        zlib1g-dev \
-    " \
-    && apt-get update && apt-get install -y --no-install-recommends $BUILD_DEPS \
-    && python3.9 -m venv /venv \
-    && /venv/bin/pip install -U pip \
-    && /venv/bin/pip install --no-cache-dir -r /requirements/production.txt \
-    && apt-get purge -y --auto-remove -o APT::AutoRemove::RecommendsImportant=false $BUILD_DEPS \
-    && rm -rf /var/lib/apt/lists/*
+RUN useradd bakerydemo --create-home && mkdir /app $VIRTUAL_ENV && chown -R bakerydemo /app $VIRTUAL_ENV
 
-RUN mkdir /code/
-WORKDIR /code/
-ADD . /code/
-EXPOSE 8000
-
-# Add custom environment variables needed by Django or your settings file here:
-ENV DJANGO_SETTINGS_MODULE=bakerydemo.settings.production DJANGO_DEBUG=off
+WORKDIR /app
 
-# Tell uWSGI where to find your wsgi file:
-ENV UWSGI_WSGI_FILE=bakerydemo/wsgi.py
+# Set default environment variables. They are used at build time and runtime.
+# If you specify your own environment variables on Heroku, they will
+# override the ones set here. The ones below serve as sane defaults only.
+#  * PATH - Make sure that Poetry is on the PATH, along with our venv
+#  * PYTHONPATH - Ensure `django-admin` works correctly.
+#  * PYTHONUNBUFFERED - This is useful so Python does not hold any messages
+#    from being output.
+#    https://docs.python.org/3.9/using/cmdline.html#envvar-PYTHONUNBUFFERED
+#    https://docs.python.org/3.9/using/cmdline.html#cmdoption-u
+#  * DJANGO_SETTINGS_MODULE - default settings used in the container.
+#  * PORT - default port used. Please match with EXPOSE.
+#    Heroku will ignore EXPOSE and only set PORT variable. PORT variable is
+#    read/used by Gunicorn.
+#  * WEB_CONCURRENCY - number of workers used by Gunicorn. The variable is
+#    read by Gunicorn.
+#  * GUNICORN_CMD_ARGS - additional arguments to be passed to Gunicorn. This
+#    variable is read by Gunicorn
+ENV PATH=$VIRTUAL_ENV/bin:$PATH \
+    PYTHONPATH=/app \
+    PYTHONUNBUFFERED=1 \
+    DJANGO_SETTINGS_MODULE=bakerydemo.settings.production \
+    PORT=8000 \
+    WEB_CONCURRENCY=3 \
+    GUNICORN_CMD_ARGS="-c gunicorn-conf.py --max-requests 1200 --max-requests-jitter 50 --access-logfile - --timeout 25 --reload"
 
-# Base uWSGI configuration (you shouldn't need to change these):
-ENV UWSGI_VIRTUALENV=/venv UWSGI_HTTP=:8000 UWSGI_MASTER=1 UWSGI_HTTP_AUTO_CHUNKED=1 UWSGI_HTTP_KEEPALIVE=1 UWSGI_UID=1000 UWSGI_GID=2000 UWSGI_LAZY_APPS=1 UWSGI_WSGI_ENV_BEHAVIOR=holy
+# Make $BUILD_ENV available at runtime
+ARG BUILD_ENV
+ENV BUILD_ENV=${BUILD_ENV}
```

Comment:
Not sure if this is actually required except for our specific use case, so could probably remove it for simplicity. If you want to keep it, move the `ENV` instruction later in the file to prevent it causing layer cache issues.